src/jit/flowgraph.cpp

```diff
@@ -22472,3 +22476,652 @@ void Compiler::fgLclFldAssign(unsigned lclNum)
+            {
+                if (block->bbJumpDest == firstBlock)
+                {
+                    BasicBlock* finallyReturnBlock  = block->bbNext;
+                    BasicBlock* postTryFinallyBlock = finallyReturnBlock->bbJumpDest;
+
+                    if (postTryFinallyBlock == normalCallFinallyReturn)
+                    {
+                        // This call returns to the expected spot, so
+                        // retarget it to branch to the clone.
+                        block->bbJumpDest = firstCloneBlock;
+                        block->bbJumpKind = BBJ_ALWAYS;
+
+                        // For some reason finally entry blocks sometimes
+                        // run a ref count deficit... sigh.
+                        // assert(firstBlock->bbRefs > 0);
+
+                        // fgRemoveRefPred(firstBlock, block);
+                        fgAddRefPred(firstCloneBlock, block);
+                    }
+                    else
+                    {
+                        // We can't retarget this call since it
+                        // returns somewhere else.
+                        retargetedAllCalls = false;
+                    }
+
+                    // Todo -- delete the paired block?
+                }
+            }
+        }
+
+        // If we retargeted all calls, modify EH descriptor to be
+        // try/fault instead of try finally, and then non-cloned
+        // finally catch type to be fault.
+        if (retargetedAllCalls)
+        {
+            JITDUMP("All callfinallies retargeted; changing finally to fault.\n");
+            HBtab->ebdHandlerType  = EH_HANDLER_FAULT;
+            firstBlock->bbCatchTyp = BBCT_FAULT;
+        }
+        else
+        {
+            JITDUMP("Some callfinallies *not* retargeted, so region must remain as a finally.\n");
+        }
+
+        // Modify first block of cloned finally to be a "normal" block.
+        BasicBlock* firstClonedBlock = blockMap[firstBlock];
+        firstClonedBlock->bbCatchTyp = BBCT_NONE;
+
+        // The normalCallFinallyReturn may be a finalStep block.
+        // It is now a normal block,so clear the special keep
+        // always flag.
+        normalCallFinallyReturn->bbFlags &= ~BBF_KEEP_BBJ_ALWAYS;
+
+#if !FEATURE_EH_FUNCLETS
+        // Remove the GT_END_LFIN from the normalCallFinallyReturn
+        // since no callfinally returns there anymore.
+        GenTreeStmt* endFinallyStmt = normalCallFinallyReturn->lastStmt();
+        GenTreePtr   endFinallyExpr = endFinallyStmt->gtStmtExpr;
+        assert(endFinallyExpr->gtOper == GT_END_LFIN);
+        fgRemoveStmt(normalCallFinallyReturn, endFinallyStmt);
+#endif
+
+        // Todo -- mark cloned blocks as a cloned finally....
+
+        // Done!
+        JITDUMP("\nDone with region %u\n\n", XTnum);
+        cloneCount++;
+    }
+
+    if (cloneCount > 0)
+    {
+        JITDUMP("fgCloneFinally() cloned %u finally handlers\n", cloneCount);
+
```

Comment:
In S.P.Corelib there were 1292 methods with try/finallys, and the jit created 1956 finally clones. There were 52 methods with empty finallys and 64 empty finallys total.